src/osyris/core/array.py

```diff
@@ -253,6 +256,24 @@ def __mul__(self, other):
                                   unit=1.0 * result.units)
         return self.__class__(values=self._array * other, unit=self._unit)
 
+    def __matmul__(self, other):
+        if isinstance(other, self.__class__):
+            scale_l = self._unit.to_base_units()
+            scale_r = other._unit.to_base_units()
+            result = scale_l * scale_r
+            lhs = self._array
+            rhs = other._array * result.magnitude
+            lhs, rhs = self._broadcast(lhs, rhs)
```

Comment:
Do you really want to broadcast here? I think we need to be careful.

Basically we really need a few unit tests to check the behaviour is what we expect.

Reply:
I'll try to come up with something that makes sure the operator overload is working properly.